src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import List, Callable, Any, Optional
+import logging
+
+from snowflake.snowpark._internal.data_source.datasource_typing import (
+    Connection,
+)
+from snowflake.snowpark.exceptions import SnowparkDataframeReaderException
+from snowflake.snowpark.types import StructType
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseDriver(ABC):
+    def __init__(self, create_connection: Callable[[], "Connection"]) -> None:
+        self.create_connection = create_connection
+
+    @abstractmethod
+    def to_snow_type(self, schema: List[Any]) -> StructType:
+        pass
+
+    @abstractmethod
+    def prepare_connection(
+        self,
+        conn: "Connection",
+        query_timeout: int = 0,
+    ) -> "Connection":
+        pass
```

Comment:
it's better to raise NotImplementedError so that future developers developing new driver won't forget to implement the two functions